src/chttp.cpp

```diff
@@ -195,6 +209,7 @@ LUA_FUNCTION(CHTTP) {
 	if (!LUA->IsType(1, Lua::Type::Table)) {
 		LOG("No HTTPRequest table set.");
 		ret = false;
+		delete request;
```

Comment:
Since we're setting `ret = false` whenever the request failed instantly, should we just move the `delete request` to the common `exit:` section? That would also take care of freeing the request when it fails to schedule (although there are currently no code paths where we check for a scheduling fail).